pkg/webhook/topic_validator.go

```diff
@@ -121,9 +145,65 @@ func (s *webhookServer) validateKafkaTopic(topic *banzaicloudv1alpha1.KafkaTopic
 		log.Info(fmt.Sprintf("Spec is requesting replication factor of %v, larger than cluster size of %v", topic.Spec.ReplicationFactor, broker.NumBrokers()))
 		return notAllowed(invalidReplicationFactorErrMsg, metav1.StatusReasonBadRequest)
 	}
+	return nil
+}
 
-	// everything looks a-okay
-	return &admissionv1beta1.AdmissionResponse{
-		Allowed: true,
+// checkExistingKafkaTopicCRs checks whether there's any other duplicate KafkaTopic CR exists
+// that refers to the same KafkaCluster's same topic
+func (s *webhookServer) checkExistingKafkaTopicCRs(ctx context.Context,
+	clusterNamespace string, topic *banzaicloudv1alpha1.KafkaTopic) *admissionv1beta1.AdmissionResponse {
+	// check KafkaTopic in the referred KafkaCluster's namespace
+	kafkaTopicList := banzaicloudv1alpha1.KafkaTopicList{}
+	err := s.client.List(ctx, &kafkaTopicList,
```

Comment:
Why can't just search for duplicate topics across all namespaces?

Reply:
We can merge it into one list query.